3.0.0.md

```diff
@@ -0,0 +1,209 @@
+      const file = new Parse.File(request.params.name, { base64: contents.buffer.toString('base64') });
+      file.save({
+        success: function() {
+          response.success(file);
+        },
+        error: response.error
+      });
+    }, 
+    error: response.error
+  });
+});
+```
+
+You would call this method with:
+
+```js
+Parse.Cloud.run({
+  url: '....',
+  name: 'my-file'
+});
+```
+
+To migrate this function you would follow the same practices as the ones before, and we'll jump right away to the async/await implementation:
+
+```js
+// after with async/await
+Parse.Cloud.beforeSave('MyClassName', async (request) => {
+  const {
+    url, name
+  } = request.params;
+  const response = await Parse.Cloud.httpRequest({ url });
+
+  const file = new Parse.File(name, { base64: response.buffer.toString('base64') });
+  await file.save();
+  return file;
+});
+```
+
+## Migrating jobs
+
+As the hooks and functions, jobs don't have a status object anymore.
+The message method was moved to the request object.
+
+```js
+// before
+Parse.Cloud.job('downloadImageJob', function(request, status) {
+  var query = new Parse.Query('ImagesToDownload');
+  query.find({
+    success: function(images) {
+      var done = 0;
+      for (var i = 0; i<images.length; i++) {
+        var image = images[i];
+        status.message('Doing '+image.get('url'));
+        Parse.Cloud.httpRequest({
+          url: image.get('url'),
+          success: function(contents) {
+            status.message('Got '+image.get('url'));
+            const file = new Parse.File(image.get('name'), { base64: contents.buffer.toString('base64') });
+            file.save({
+              success: function() {
+                status.message('Saved '+image.get('url'));
+                done++;
+                if (done == images.length) {
+                  status.success();
+                }
+              },
+              error: function() {
+                response.message('Error '+image.get('url'));
+                done++;
+                if (done == images.length) {
+                  status.error();
+                }
+              }
+            });
+          }, 
+          error: status.error
+        });
+      }
+    }
+  });
+});
+```
+
+```js
+Parse.Cloud.job('downloadImageJob', async function(request) {
+  const query = new Parse.Query('ImagesToDownload');
+  const images = await query.find();
+  const promises = images.map(async (image) => {
+    request.message('Doing ' + image.get('url'));
+    const contents = await Parse.Cloud.httpRequest({
+      url: image.get('url')
+    });
+    request.message('Got ' + image.get('url'));
+    const file = new Parse.File(image.get('name'), { base64: contents.buffer.toString('base64') });
+    await file.save();
+    request.message('Saved ' + image.get('url'));
+  });
+  await Promise.all(promises);
+});
+
+```
+
+As you can see the new implementation is way more concise and easier to read and maintain.
```

Comment:
As you can see, the new implementation is more concise, easier to read and maintain.